streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -745,9 +780,21 @@ protected void maybeCommit(final long now) {
      * Commit the states of all its tasks
      */
     private void commitAll() {
-        for (final StreamTask task : activeTasks.values()) {
-            commitOne(task);
+        final RuntimeException e = performOnStreamTasks(new StreamTaskAction() {
+            @Override
+            public String name() {
+                return "commit";
+            }
+
+            @Override
+            public void apply(final StreamTask task) {
+                commitOne(task);
+            }
+        });
+        if (e != null) {
+            throw e;
```

Comment:
As discussed offline. This code behaves as before. I will do an separate PR to improve exception handling in general.